### PR TITLE
Fix: refactor flakey category dedupe test

### PR DESCRIPTION
**Pull request by bsardo**

### CODE REVIEW

**Review by VeronikaSolovei9 — COMMENTED** [review]
Looks good, left one minor suggestion.

File: exchange/exchange_test.go [hunk 1]
```diff
@@ -2974,84 +2974,89 @@ func TestCategoryMappingTranslateCategoriesFalse(t *testing.T) {
...
+				"bid_id3": "20.00_Electronics_30s",
+			},
+		},
 	}
 
-	numIterations := 10
-
-	// Run the function many times, this should be enough for the 50% chance of which bid to remove to remove bid1 sometimes
-	// and bid3 others. It's conceivably possible (but highly unlikely) that the same bid get chosen every single time, but
-	// if you notice false fails from this test increase numIterations to make it even less likely to happen.
```
Comment (VeronikaSolovei9) [inline exchange/exchange_test.go:3021]: Was I drunk when I wrote it?

File: exchange/exchange_test.go [hunk 2]
```diff
@@ -2974,84 +2974,89 @@ func TestCategoryMappingTranslateCategoriesFalse(t *testing.T) {
...
+						&bid1_1,
+						&bid1_2,
+						&bid1_3,
+						&bid1_4,
+						&bid1_5,
+					},
+					Currency: "USD",
+				},
+			}
+			deduplicateGenerator := DeduplicateBidBooleanGenerator{value: tt.dedupeGeneratorValue}
```
[the remark this answers is not in the document]
Comment (AlexBVolcy) [reply exchange/exchange_test.go:3051]: I replaced the generators and the tests still passed, so I think we can use the old one!

**Review by AlexBVolcy — COMMENTED** [review]
This looks good, just left one comment regarding the generator usage!

File: exchange/exchange_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (VeronikaSolovei9) [reply exchange/exchange_test.go:3051]: Yes, I tried it too :)

**Review by VeronikaSolovei9 — APPROVED** [review]

**Pull request merged**